### PR TITLE
StorageLocationEvent

### PR DESCRIPTION
**Pull request by pdowler**
model
libraries
critwall
fenwick

### CODE REVIEW

File: fenwick/src/intTest/java/org/opencadc/fenwick/StorageLocationEventSyncTest.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+            Assert.assertNotNull(r2);
+            Assert.assertTrue(r2.siteLocations.contains(sloc1));
+            Assert.assertTrue(r2.siteLocations.contains(sloc2));
+
+            Artifact r3 = inventoryEnvironment.artifactDAO.get(a3.getID());
+            Assert.assertNotNull(r3);
+            Assert.assertTrue(r3.siteLocations.contains(sloc2));
+
+            StorageLocationEvent actual = inventoryEnvironment.storageLocationEventDAO.get(sle2.getID());
+            Assert.assertNull(actual);
```
Comment (jburke-cadc) [inline fenwick/src/intTest/java/org/opencadc/fenwick/StorageLocationEventSyncTest.java:240]: an assert that sle1 isn't null?

File: fenwick/src/main/java/org/opencadc/fenwick/StorageLocationEventSync.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    }
+    
+    @Override
+    void doit() throws ResourceNotFoundException, IOException, IllegalStateException, TransientException, InterruptedException {
+        final MessageDigest messageDigest;
+        try {
+            messageDigest = MessageDigest.getInstance("MD5");
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("BUG: failed to get instance of MD5", e);
+        }
```
Comment (jburke-cadc) [inline fenwick/src/main/java/org/opencadc/fenwick/StorageLocationEventSync.java:120]: messageDisgest isn't used.

File: fenwick/src/main/java/org/opencadc/fenwick/StorageLocationEventSync.java [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+        final SiteLocation remoteSiteLocation = (storageSite == null ? null : new SiteLocation(storageSite.getID()));
```
Comment (jburke-cadc) [inline fenwick/src/main/java/org/opencadc/fenwick/StorageLocationEventSync.java:122]: remoteSiteLocation isn't used.

File: fenwick/src/main/java/org/opencadc/fenwick/StorageLocationEventSync.java [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+        
+        final HarvestState harvestState = this.harvestStateDAO.get(StorageLocationEvent.class.getName(), resourceID);
+
+        final Date endTime = new Date();
+        final Date lookBack = new Date(endTime.getTime() - LOOKBACK_TIME);
+        Date startTime = getQueryLowerBound(lookBack, harvestState.curLastModified);
+        
+        DateFormat df = DateUtil.getDateFormat(DateUtil.IVOA_DATE_FORMAT, DateUtil.UTC);
+        if (lookBack != null && harvestState.curLastModified != null) {
```
Comment (jburke-cadc) [inline fenwick/src/main/java/org/opencadc/fenwick/StorageLocationEventSync.java:131]: lookBack cannot be null.

File: fenwick/src/main/java/org/opencadc/fenwick/StorageLocationEventSync.java [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+            log.debug("lookBack=" + df.format(lookBack) + " curLastModified=" + df.format(harvestState.curLastModified) 
+                + " -> " + df.format(startTime));
+        }
+        String start = null;
+        if (startTime != null) {
+            start = df.format(startTime);
+        }
+        String end = null;
+        if (endTime != null) {
```
Comment (jburke-cadc) [inline fenwick/src/main/java/org/opencadc/fenwick/StorageLocationEventSync.java:140]: endTime cannot be null.

File: fenwick/src/main/java/org/opencadc/fenwick/StorageLocationEventSync.java [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+                   InterruptedException {
+        final String query = buildQuery(start, end);
+        log.debug("adql: " + query);
+        return tapClient.query(query, new StorageLocationEventRowMapper());
+    }
+    
+    private String buildQuery(Date startTime, Date endTime) {
+        DateFormat df = DateUtil.getDateFormat(DateUtil.ISO_DATE_FORMAT, DateUtil.UTC);
+        
+        StringBuilder sb = new StringBuilder();
```
Comment (jburke-cadc) [inline fenwick/src/main/java/org/opencadc/fenwick/StorageLocationEventSync.java:219]: sb is not used.

**Review by jburke-cadc — CHANGES_REQUESTED** [review]
Just a few things to cleanup.

File: fenwick/src/main/java/org/opencadc/fenwick/StorageLocationEventSync.java (hunk 2 above)
Comment (pdowler) [reply fenwick/src/main/java/org/opencadc/fenwick/StorageLocationEventSync.java:120]: Oh, actually it should be used to check that the event is intact... and so should the two Deleted??Sync classes. 
That was missing from day one (and I reviewed that code :-) Adding those checks.

**Pull request merged**